### ISSUE TITLE
Makefile failure in v2.3.0

### ISSUE THREAD

**Issue by AviManor** [aws/aws-cdi-sdk#62]
Regarding AWS CDI-SDK Release v2.3.0
Running the Makefile ends with and error:
`make: command: Command not found`
`make: command: Command not found`
`Makefile:253: *** CMake not found. See INSTALL instructions..  Stop.`

**To solve the problem I have replaced the lines:**
`# Determine CMake executable`
`ifneq ($(shell command -v cmake3),)`
`    CMAKE := $(shell command -v cmake3)`
`else`
`    CMAKE := $(shell command -v cmake)`
`endif`

`ifeq ($(CMAKE),)`
`    $(error CMake not found. See INSTALL instructions.)`
`endif`

**With the lines:**
`# Determine CMake executable`
`ifneq ($(shell which cmake3),)`
`    CMAKE := $(shell which cmake3)`
`else`
`    CMAKE := $(shell which cmake)`
`endif`

`$(info CMAKE $(CMAKE))`
`ifeq ($(CMAKE),)`
`    $(error CMake not found. See INSTALL instructions.)`
`endif`

**Comment by AviManor** [comment]
Had to add **sudo** because of permissions issues:

`# rule to create the various build output directories`
`$(foreach d,obj lib bin doc packages libfabric results image libaws,$(build_dir.$(d))) :`
`	$(Q)sudo mkdir -p $@`

**Comment by ralfjuen** [comment]
> Regarding AWS CDI-SDK Release v2.3.0
> Running the Makefile ends with and error:
> `make: command: Command not found`
> `make: command: Command not found`
> `Makefile:253: *** CMake not found. See INSTALL instructions.. Stop.`

Hi Avi,

Thanks for reporting this issue. To help us provide a solution, please let me know what you see when typing the following commands in your shell:

`cat /etc/*release`
`make -v`
`command -v cmake`
`command -v cmake3`

**Comment by AviManor** [comment]
Hi
The "command" runs well in the shell, but not in the makefile.
Did you notice the "**sudo** mkdir -p" issue?

ubuntu@ip-172-31-0-174:/home/AviManor/aws-cdi-sdk$ cat /etc/*release
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=16.04
DISTRIB_CODENAME=xenial
DISTRIB_DESCRIPTION="Ubuntu 16.04.7 LTS"
NAME="Ubuntu"
VERSION="16.04.7 LTS (Xenial Xerus)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 16.04.7 LTS"
VERSION_ID="16.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
VERSION_CODENAME=xenial
UBUNTU_CODENAME=xenial
ubuntu@ip-172-31-0-174:/home/AviManor/aws-cdi-sdk$ make -v
GNU Make 4.1
Built for x86_64-pc-linux-gnu
Copyright (C) 1988-2014 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
ubuntu@ip-172-31-0-174:/home/AviManor/aws-cdi-sdk$ command -v cmake
/usr/bin/cmake
ubuntu@ip-172-31-0-174:/home/AviManor/aws-cdi-sdk$ command -v cmake3
ubuntu@ip-172-31-0-174:/home/AviManor/aws-cdi-sdk$

**Comment by ralfjuen** [comment]
> Had to add **sudo** because of permissions issues:
> 
> `# rule to create the various build output directories`
> `$(foreach d,obj lib bin doc packages libfabric results image libaws,$(build_dir.$(d))) :`
> ` $(Q)sudo mkdir -p $@`

This change is not necessary if you execute `make` as a unix user which has permissions to create directories and files in your working directory.  The Makefile does not contain instructions to check these permissions as it is customary to choose a working directory where you have full write permissions.

**Comment by AviManor** [comment]
Hi
I have followed the installation instructions on the instance from scratch. I SSH to the instance as user "ubuntu".
The permissions should be correct.
Please review the installation instructions and see it I have made something wrong.
Avi

**Comment by ralfjuen** [comment]
Thanks for the information. We are working on a fix for the CMake issue. I will leave this github issue open until the fix is published.

The permissions issue does not look like a problem with our build process but with insufficient write permissions. Please try building in a new directory where your unix user has full write permissions. If the issue persists, please create a new github issue with more details (error message, steps to reproduce).

**Comment by f2bo** [comment]
I'm also having the same CMake issue. Note that even though running `command -v cmake` in my system does return `/usr/bin/cmake`, it seems that the following lines do not initialize the CMAKE variable as expected. 

```
# Determine CMake executable
ifneq ($(shell command -v cmake3),)
    CMAKE := $(shell command -v cmake3)
else
    CMAKE := $(shell command -v cmake)
endif
```

I patched the Makefile to hard-code the CMAKE variable. However, there seems to be a second problem which prevents building since I also see the following error. 

```
CMake Error: Unknown argument -j
CMake Error: Run 'cmake --help' for all supported options.
```
This happens with **CMake version 3.21.1**. I was able to build the SDK after removing the `-j $$(nproc)` argument from the CMake command line (see https://gitlab.kitware.com/cmake/cmake/-/issues/21978).

**Comment by ralfjuen** [comment]
> However, there seems to be a second problem which prevents building since I also see the following error.
> 
> ```
> CMake Error: Unknown argument -j
> CMake Error: Run 'cmake --help' for all supported options.
> ```

Thank you for reporting this. We will include a fix for this issue in the next release.

**Comment by ralfjuen** [comment]
Fixed in v2.3.1.
